src/CsharpClient/QuixStreams.Streaming/States/StreamStateManager.cs

```diff
@@ -0,0 +1,235 @@
+        /// <returns>The newly created <see cref="StreamState{T}"/> instance.</returns>
+        private StreamState<T> CreateStreamState<T>(string nameOfState, StreamStateDefaultValueDelegate<T> defaultValueFactory = null)
+        {
+            this.logger.LogTrace("Creating Stream state for {0}", nameOfState);
+            var state = new StreamState<T>(this.stateStorage.GetOrCreateSubStorage(nameOfState), defaultValueFactory, new PrefixedLoggerFactory(this.loggerFactory, $"{logPrefix} - {nameOfState}"));
+            return state;
+        }
+        
+        /// <summary>
+        /// Creates a new instance of the <see cref="StreamState"/> class with the specified <paramref name="nameOfState"/>.
+        /// </summary>
+        /// <param name="nameOfState">The name of the state.</param>
+        /// <returns>The newly created <see cref="StreamState"/> instance.</returns>
+        private StreamState CreateStreamState(string nameOfState)
+        {
+            this.logger.LogTrace("Creating Stream state for {0}", nameOfState);
+            var state = new StreamState(this.stateStorage.GetOrCreateSubStorage(nameOfState), new PrefixedLoggerFactory(this.loggerFactory, $"{logPrefix} - {nameOfState}"));
+            return state;
+        }
+        
+        /// <summary>
+        /// Returns an enumerable collection of all available state names for the current stream.
+        /// </summary>
+        /// <returns>An enumerable collection of string values representing the names of the stream states.</returns>
+        public IEnumerable<string> GetStates()
+        {
+            return this.stateStorage.GetSubStorages();
+        }
+        
+        /// <summary>
+        /// Deletes all states for the current stream.
+        /// </summary>
+        /// <returns>The number of states that were deleted.</returns>
+        public int DeleteStates()
+        {
+            this.logger.LogTrace("Deleting Stream states for {0}", streamId);
+            var count = this.stateStorage.DeleteSubStorages();
+            this.states.Clear();
+            return count;
+        }
+        
+        /// <summary>
+        /// Deletes the state with the specified name
+        /// </summary>
+        /// <returns>Whether the state was deleted</returns>
+        public bool DeleteState(string nameOfState)
+        {
+            this.logger.LogTrace("Deleting Stream state {0} for {1}", nameOfState, streamId);
+            if (!this.stateStorage.DeleteSubStorage(nameOfState)) return false;
+            this.states.Remove(nameOfState);
+            return true;
+        }
+        
+        /// <summary>
+        /// Creates a new application state with automatically managed lifecycle for the stream
+        /// </summary>
+        /// <param name="nameOfState">The name of the state</param>
+        /// <returns>Stream state</returns>
+        public StreamState GetState(string nameOfState)
```

Comment:
maybe rename `nameOfState` to `stateName` if it is not wrong.
nameOfState is mentioned multiple times in this file